packages/contracts-bridge/contracts/test/utils/BridgeTest.sol

```diff
@@ -83,26 +91,47 @@ contract BridgeTest is Test {
             address(bridgeToken),
             address(upgradeBeaconController)
         );
+        // Tests that concern the EThereumBridgeRouter, they will already have called
+        // setUpEthereumBridgeRouter(), which means that the address will not be address(0)
+        if (address(bridgeRouter) == address(0)) {
+            setUpBridgeRouter();
+        }
         initializeContracts();
+        initializeBridgeRouter();
         remoteTokenLocalAddress = createRemoteToken(
             remoteDomain,
             remoteTokenRemoteAddress
         );
     }
 
-    function initializeContracts() public {
-        tokenRegistry.initialize(
-            address(tokenBeacon),
-            address(xAppConnectionManager)
+    function setUpEthereumBridgeRouter() public {
+        mockAccountant = new MockAccountant();
+        bridgeRouter = IBridgeRouterHarness(
+            address(new EthereumBridgeRouterHarness(address(mockAccountant)))
         );
+    }
+
+    function setUpBridgeRouter() public {
+        bridgeRouter = IBridgeRouterHarness(address(new BridgeRouterHarness()));
+    }
+
+    function initializeBridgeRouter() public virtual {
         bridgeRouter.initialize(
             address(tokenRegistry),
             address(xAppConnectionManager)
         );
         // The Token Registry should be owned by the Bridge Router
+        bridgeRouter.enrollRemoteRouter(remoteDomain, remoteBridgeRouter);
+    }
+
+    function initializeContracts() public virtual {
```

Comment:
why is bridge router separated now? Either each should be separate, with `initializeContracts` doing all 3, or they should all be under `initializeContracts`